### PR TITLE
Implemented Revelation Dance

### PR DESCRIPTION
**Pull request by lucfd**
~~- Added 'Typeless' type~~
~~- has neutral attacking & defensive matchups~~
~~- is assigned to Pokemon that have no types~~
  ~~-is removed from Pokemon that gain a type~~
- Created new attribute `MatchUserTypeAttr` that changes a move's typing to match the user's primary type.
- Implemented Revelation Dance.
  - Matches the user's primary type
  - Can become typeless
  - Changes to your active tera type (except for stellar, in which case it uses your previous type)
- Fixed Steam Engine activating on status moves.

### CODE REVIEW

**Comment by Xavion3** [comment]
This implementation is incomplete, Revelation Dance should be able to become typeless if the user is typeless somehow, but also Revelation Dance shouldn't change type if the user is tera stellar, and currently it looks like it would become stellar type.

**Comment by lucfd** [comment]
Good catch on the stellar type behaviour. Revelation Dance will work with typeless once it's implemented (presumably it will be another type with neutral matchups across the board), but I guess now's as good a time as ever to add it

**Comment by lucfd** [comment]
I believe so. I'm going off of this video, which is weirdly the only thing I can find on this edge case https://www.youtube.com/watch?v=sX23y80fYw8

**Comment by Flashfyre** [comment]
There's a chance this will crash somewhere because there's no type image for it. I will need to test this.

**Comment by lucfd** [comment]
In my testing it pulled a question mark image which I presume is a default case

**Comment by Procyonae** [comment]
You could make stuggle use your new typeless type too so there's not two ways of doing it simultaneously

**Comment by lucfd** [comment]
I've removed `Type.TYPELESS` completely and switched everything to the `Type.UNKNOWN` implementation agreed upon in Discord. Let me know if any other changes need to be made!

**Review by Xavion3 — COMMENTED** [review]
This looks generally fine though, just those few small things.

File: src/data/type.ts [hunk 1]
```diff
@@ -543,4 +543,4 @@ export function getTypeRgb(type: Type): [ integer, integer, integer ] {
     default:
       return [ 0, 0, 0 ];
   }
-}
\ No newline at end of file
+}
```
[the remark this answers is not in the document]
Comment (lucfd) [reply src/data/type.ts:546]: This diff is caused by the addition of an EOF newline character

**Comment by lucfd** [comment]
Debug statement removed & test conditions have been simplified

**Comment by chacolah** [comment]
Tested with fire Oricorio after rebasing current mainbranch, no issues, works as intended.

**Review by Madmadness65 — APPROVED** [review]

**Review by Xavion3 — COMMENTED** [review]
Looks good now
